File: app/src/main/java/com/recursia/popularmovies/data/mappers/CastDatabaseModelToEntityMapper.java
package com.recursia.popularmovies.data.mappers;

import com.recursia.popularmovies.data.models.CastDatabaseModel;
import com.recursia.popularmovies.domain.models.Cast;
import com.recursia.popularmovies.utils.Mapper;

public class CastDatabaseModelToEntityMapper extends Mapper<CastDatabaseModel, Cast> {
    @Override
    public Cast transform(CastDatabaseModel castDatabaseModel) {
        Cast cast = new Cast();
        cast.setCastId(castDatabaseModel.getCastId());
        cast.setCharacter(castDatabaseModel.getCharacter());
        cast.setCreditId(castDatabaseModel.getCreditId());
        cast.setGender(castDatabaseModel.getGender());
        cast.setId(castDatabaseModel.getId());
        cast.setProfilePath(castDatabaseModel.getProfilePath());
        cast.setMovieId(castDatabaseModel.getMovieId());
        cast.setName(castDatabaseModel.getName());
        cast.setOrder(castDatabaseModel.getOrder());
        return cast;
    }
}

File: app/src/main/java/com/recursia/popularmovies/data/mappers/CastDatabaseModelToEntityMapper.kt
package com.recursia.popularmovies.data.mappers

import com.recursia.popularmovies.data.models.CastDatabaseModel
import com.recursia.popularmovies.domain.models.Cast
import com.recursia.popularmovies.utils.Mapper

class CastDatabaseModelToEntityMapper : Mapper<CastDatabaseModel, Cast>() {
    override fun transform(castDatabaseModel: CastDatabaseModel): Cast {
        val cast = Cast()
        cast.castId = castDatabaseModel.castId
        cast.character = castDatabaseModel.character
        cast.creditId = castDatabaseModel.creditId
        cast.gender = castDatabaseModel.gender
        cast.id = castDatabaseModel.id
        cast.profilePath = castDatabaseModel.profilePath
        cast.movieId = castDatabaseModel.movieId
        cast.name = castDatabaseModel.name
        cast.order = castDatabaseModel.order
        return cast
    }
}
